src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaContractTests.cs

```diff
@@ -197,6 +231,366 @@ public static void Dispose_CallsVirtual(SlhDsaAlgorithm algorithm)
             SlhDsaTestHelpers.VerifyDisposed(slhDsa);
         }
 
+        [Theory]
+        [MemberData(nameof(SlhDsaTestData.AlgorithmsData), MemberType = typeof(SlhDsaTestData))]
+        public static void ExportPkcs8PrivateKey_CallsExportSlhDsaSecretKey(SlhDsaAlgorithm algorithm)
+        {
+            SlhDsaTestHelpers.AssertExportPkcs8PrivateKey(export =>
+            {
+                using SlhDsaMockImplementation slhDsa = SlhDsaMockImplementation.Create(algorithm);
+
+                slhDsa.ExportSlhDsaSecretKeyCoreHook = _ => { };
+                slhDsa.AddLengthAssertion();
+                slhDsa.AddFillDestination(1);
+
+                // SlhDsaMockImplementation overrides TryExportPkcs8PrivateKeyCore with a stub. In order to replicate the
+                // non-overridden behavior, we will replace the stub with a call to base.TryExportPkcs8PrivateKeyCore.
+                // We can then assert that base.TryExportPkcs8PrivateKeyCore calls ExportSlhDsaSecretKeyCore as expected.
+                slhDsa.TryExportPkcs8PrivateKeyCoreHook = slhDsa.BaseTryExportPkcs8PrivateKeyCore;
+
+                // Invoke the export
+                byte[] exported = export(slhDsa);
+
+                // Assert that the core methods were called
+                AssertExtensions.GreaterThan(slhDsa.ExportSlhDsaSecretKeyCoreCallCount, 0);
+                AssertExtensions.GreaterThan(slhDsa.TryExportPkcs8PrivateKeyCoreCallCount, 0);
+
+                // And check the returned data
+                PrivateKeyInfoAsn exportedPkcs8 = PrivateKeyInfoAsn.Decode(exported, AsnEncodingRules.DER);
+                AssertExtensions.SequenceEqual(CreateFilledArray(algorithm.SecretKeySizeInBytes, 1), exportedPkcs8.PrivateKey.Span);
+                Assert.Equal(0, exportedPkcs8.Version);
+                Assert.Equal(SlhDsaTestHelpers.AlgorithmToOid(algorithm), exportedPkcs8.PrivateKeyAlgorithm.Algorithm);
+                Assert.Equal(default(ReadOnlyMemory<byte>?), exportedPkcs8.PrivateKeyAlgorithm.Parameters);
```

Comment:
```suggestion
                AssertExtensions.FalseExpression(exportedPkcs8.PrivateKeyAlgorithm.Parameters.HasValue);
```

I think that'll be a lot clearer failure message if it ever comes up.